### PR TITLE
107 fix bug with scrolling categories in signup flow

### PR DESCRIPTION
**Pull request by Mathurinnn**
Fixed display problem for interests in SetUpProfile

### CODE REVIEW

File: app/src/main/java/com/github/se/gatherspot/ui/SetUpProfile.kt
Comment (WesLeeO) [inline app/src/main/java/com/github/se/gatherspot/ui/SetUpProfile.kt:1]: Ok, I think this is a good way to solve the bug. I suppose when an interest that is selected gets out of the view and reappears, this causes the composable to be recomposed and thus the selected is initialized to false even though being in the lists. This solves the issue by adding another predicate that makes selected true in the right cases.

File: app/src/androidTest/java/com/github/se/gatherspot/authentification/SetUpTest.kt
Comment (WesLeeO) [inline app/src/androidTest/java/com/github/se/gatherspot/authentification/SetUpTest.kt:1]: I'm not sure how this test makes sure interests do not get deselected automatically

**Review by WesLeeO — COMMENTED** [review]
Overall, good just explain me how the test makes sure the bug is not present and it will good to merged

**Comment by Mathurinnn** [comment]
now the tests make sure it will not happen again. Basically we leave the first interest selected and we come back to it at the end of the scrolling and assert that is not in the add position which makes it unselected

**Review by WesLeeO — APPROVED** [review]

**Pull request merged**